### PR TITLE
Add Test Case for Special Character Paths

### PR DESCRIPTION
**Pull request by schrieveslaach**
Added test cases that ensure that special characters in path names are passed to external commands correctly. These cases have been implemented with rstest to reuse existing test code.

### CODE REVIEW

File: tests/shell/pipeline/commands/external.rs [hunk 1]
```diff
@@ -219,21 +219,29 @@ mod external_words {
         assert_eq!(actual.out, "test \"things\"");
     }
 
-    #[test]
-    fn external_arg_with_quotes() {
+    #[rstest::rstest]
+    #[case("sample.toml", r#""sample.toml""#)]
+    #[case("a sample file.toml", r#""a sample file.toml""#)]
+    #[case("quote'mark.toml", r#""quote'mark.toml""#)]
+    // TODO: #[case(r#"quote"mark.toml"#, r#""quote\\"mark.toml""#)]
```
Comment (schrieveslaach) [inline tests/shell/pipeline/commands/external.rs:226]: Is it possible in nushell to escape quotes?

Comment (fdncred) [reply tests/shell/pipeline/commands/external.rs:226]: I just submitted this PR https://github.com/nushell/nushell/pull/3601 that will allow you to do `char single_quote` or `char double_quote` to make it easier to add quotation marks in nushell. single_quote can also be abbreviated as `squote` or `sq` and double_quote can be abbreviated `dquote` or `dq`.

Comment (schrieveslaach) [reply tests/shell/pipeline/commands/external.rs]: @fdncred, could you give me an example code snippet that explains how to escape quotes in nushell? I thought that I can just enable the test case but I get following error:

<pre><font color="#F66151"><b>error</b></font><b>: Unexpected end of input</b>
  <font color="#12488B">┌─</font> shell:5:47
  <font color="#12488B">│</font>  
<font color="#12488B">5</font> <font color="#12488B">│</font>                               nu --testbin meow <font color="#C01C28">&quot;quote\&quot;mark.toml&quot; | from toml | get nu_party_venue</font>
  <font color="#12488B">│</font> <font color="#C01C28">╭───────────────────────────────────────────────^</font>
<font color="#12488B">6</font> <font color="#12488B">│</font> <font color="#C01C28">│</font> 
  <font color="#12488B">│</font> <font color="#C01C28">╰^ Expected &quot;</font>

</pre>

Comment (fdncred) [reply tests/shell/pipeline/commands/external.rs]: There is an example of `meow` with quotes in the source code though.
```rust
let actual = nu!(
    cwd: dirs.test(), pipeline(
    r#"
        nu --testbin meow "sample.toml" | from toml | get nu_party_venue
    "#
));
```

Comment (schrieveslaach) [reply tests/shell/pipeline/commands/external.rs]: That's the test case that I'm extending in this PR. :wink: 

I was looking for a way to escape the quotes within a quoted argument. Like this:

```rust
let actual = nu!(
    cwd: dirs.test(), pipeline(
    r#"
        nu --testbin meow "quote\"mark.toml" | from toml | get nu_party_venue
    "#
));
```

@jonathandturner, is it correct that nushell cannot escape quotes like this and it is only possible to use following escape patch:

```
nu --testbin meow $"quote(char double_quote)mark.toml.toml" | from toml | get nu_party_venue
```

That's a lot to type for a user.

Comment (schrieveslaach) [reply tests/shell/pipeline/commands/external.rs]: @jonathandturner, do you know if `nu --testbin meow $"quote(char double_quote)mark.toml.toml" | from toml | get nu_party_venue` should work? The test case fails and I'm not sure if it is an error of in the test case or in the production code. Do you mind to have a look?

Comment (schrieveslaach) [reply tests/shell/pipeline/commands/external.rs]: Seems like a bug  #3668

**Comment by schrieveslaach** [comment]
Rebased and tests are working. I think this PR is ready for review

**Comment by jonathandturner** [comment]
Looks good!

**Pull request merged**